lib/dns/dns.go

```diff
@@ -356,34 +398,31 @@ func (s *Server) HandleVIC(w mdns.ResponseWriter, r *mdns.Msg) (bool, error) {
 	if e == nil {
 		return false, fmt.Errorf("Could not find requesting container with ip %s", ip)
 	}
-	scope := e.Scope()
 
-	if domain != "" && scope.Name() != domain {
-		return false, fmt.Errorf("Inter-scope request for container %s in %s from %s", name, domain, scope.Name())
+	if domain != "" && e.Scope().Name() != domain {
```

Comment:
`domain` can be optional, so we need to skip this check if it is not provided in the query.